### ISSUE TITLE
Why does the 'loss=nan' appear when finetuning model using fp16 or bf16 mixed precision in 'fine_tune.py' ?

### ISSUE THREAD

**Issue by zakinp** [kohya-ss/sd-scripts#135]
Same as title. This problem has existed since the v9 version. It is related to the GPU type. It appears in 3090, A40, but doesn't appear in A100.

**Comment by abhiishekpal** [comment]
Facing the same issue, were you able to resolve it? Not sure if this is xformers issue or some specific torch version

**Comment by Ebycow** [comment]
It may be a different topic or a separate issue, but when learning in the same way with multi-GPU (`accelerate launch --multi_gpu`) , NaN is displayed and the model file is corrupted.

**Comment by FlyHighest** [comment]
Try disabling xformers, or changing another version. I 've never got NAN since disabling xformers.

**Comment by litaotju** [comment]
It's a known issue of xformers library on 30x GPUs, try install the dev version of the xformers. 
This works on my 3060.
```
pip install xformers==0.0.17.dev466
```

See https://github.com/facebookresearch/xformers/issues/628

**Comment by cooperdk** [comment]
> It's a known issue of xformers library on 30x GPUs, try install the dev version of the xformers. This works on my 3060.
> 
> ```
> pip install xformers==0.0.17.dev466
> ```
> 
> See [facebookresearch/xformers#628](https://github.com/facebookresearch/xformers/issues/628)

I have this problem, but that release of xformers is not available.

**Comment by cooperdk** [comment]
They don't, the issue lies elsewhere. The error also appears without using xformers (3060 OC, py3.10.9, torch 1.12.1 and 2.x, and also from basic installation defaults).

Also, bitsandbytes is not mature enough for use, someone made a report on it, it should be avoided and might potentially be the cause of the error.

But it may also be that your training command is actually invalid. The script should check for that (and it should also support stopping after X steps or (by way of an argument) calculate values to not exceed 3000-3500 steps for LORA training since no more is recommended.

**Comment by jacquesfeng123** [comment]
I also got this problem on 4090, I am using torch 2.0. 

So I assume transformer is not use, therefore the fix above should not help.

Model is fine though. Also I would be interested into looking at some loss/validation, currently val is not supported.

I'm on commit  feb6728762a8f463d15ba936d189d4c3abfaa1ab
